nearcore/src/state_sync.rs

```diff
@@ -241,101 +237,109 @@ async fn state_sync_dump(
                 match in_progress_data {
                     Err(error) => Err(error),
                     Ok((state_root, num_parts, sync_prev_hash)) => {
-                        let missing_parts = get_missing_part_ids_for_epoch(
+                        let new_epoch_state = check_new_epoch(
```

Comment:
Please consider getting the current epoch_id at the beginning of the iteration, __before__ doing `match progress`. Doing it once should simplify things, in comparison to doing it in every branch of the `match` expression.